uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-leave-absences-manager-leave-requests.js

```diff
@@ -0,0 +1,69 @@
+var Promise = require('es6-promise').Promise;
+var page = require('./page');
+
+module.exports = (function () {
+
+  return page.extend({
+    /**
+     * Wait for the page to be ready as it waits for the actions of the first
+     * row of leave requests to be visible
+     * @return {Object} this object
+     */
+    waitForReady: function () {
+      var casper = this.casper;
+
+      casper.then(function () {
+        casper.waitUntilVisible('tbody  tr:nth-child(1) a');
+      });
+
+      return this;
+    },
+    /**
+     * Opens the given menu item from side menu
+     * @param {String} menuIndex
+     * @return {Object} this object
+     */
+    openSideMenu: function (menuIndex) {
+      var casper = this.casper;
+
+      casper.then(function () {
+        casper.waitUntilVisible('ul.chr_vertical_tabs li:nth-child(1) a', function () {
+          casper.click('ul.chr_vertical_tabs li:nth-child(' + menuIndex + ') a');
+          casper.waitUntilVisible('ul.chr_vertical_tabs li:nth-child(1) a');
+        });
+      });
+
+      return this;
+    },
+    /**
+     * Opens the dropdown for manager actions like edit/respond, cancel.
+     * @param {Number} row number corresponding to leave request in the list
+     * @return {Object} this object
+     */
+    openActionsForRow: function (row) {
+      var casper = this.casper;
+      // || was not working in string so created another variable here
+      var selectedRow = row || 1;
+
+      casper.then(function () {
+        casper.click('tr:nth-child('+ selectedRow +') > td > div > a');
```

Comment:
this selector it's too generic (chaining a generic `td` and a generic `div` tag), can't you make it more specific? Aren't there classes that you can use?